### PR TITLE
Upgrade to badger/v3

### PR DESCRIPTION
**Pull request by kelindar**
This upgrades to `badger/v3` which has a lot of memory-related changes under the hood. See https://github.com/dgraph-io/badger/blob/master/CHANGELOG.md

Noteworthy PRs:

1. https://github.com/dgraph-io/badger/pull/1555
2. https://github.com/dgraph-io/badger/pull/1574/files

### CODE REVIEW

File: internal/storage/disk/disk.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ import (
 	"sync/atomic"
 	"time"
 
-	"github.com/dgraph-io/badger/v2"
+	"github.com/dgraph-io/badger/v3"
```
Comment (ocassetti) [inline internal/storage/disk/disk.go:17]: Will it auto upgrade a v2 database on opening?

Comment (kelindar) [reply internal/storage/disk/disk.go:17]: I'm not sure, haven't tried. I would also like @qiaowei-g to review the options. There's many changes in badger and apparently they are using manual memory allocation to use a bit less memory and free it better.

Comment (qiaowei-g) [reply internal/storage/disk/disk.go:17]: based on the description in [badger-#1555](https://github.com/dgraph-io/badger/pull/1555), v2 data cannot be read by v3, so we will have to handle the deployment carefully to ensure no data loss.

**Review by qiaowei-g — APPROVED** [review]

File: internal/storage/disk/disk.go (hunk 1 above)
Comment (kelindar) [reply internal/storage/disk/disk.go:17]: Alright, I'll mark it as 1.3 release so we don't mix things up.

**Pull request merged**